### PR TITLE
resend request in case server was disconnected after request was sent

### PR DESCRIPTION
**Pull request by darkdarkdragon**

### CODE REVIEW

**Comment by wltsmrz** [comment]
Shouldn't this already occur? https://github.com/ripple/ripple-lib/blob/50acc4c7087bcef6ac9c8a25a0829f926b6b5d58/src/core/remote.js#L769-L771

**Comment by wltsmrz** [comment]
Oh nevermind, I see what you're doing now

**Comment by wltsmrz** [comment]
Might it be preferred to add a `response` event or so in [this neighborhood](https://github.com/ripple/ripple-lib/blob/50acc4c7087bcef6ac9c8a25a0829f926b6b5d58/src/core/server.js#L673) and moving this reconnect logic to request.request() or remote.request()?

**Comment by darkdarkdragon** [comment]
@wltsmrz I've moved logic into request.request

**Comment by wltsmrz** [comment]
It would be great while you're at it to allow `request.request(callback);` again. This was broken a while back. Aside from that, squash commits & LGTM

File: src/core/server.js [hunk 1]
```diff
@@ -686,6 +687,7 @@ Server.prototype._handleResponse = function(message) {
       error_message: 'Remote reported an error.',
       remote: message
     });
+    request.emit('response');
```
Comment (wltsmrz) [inline src/core/server.js]: I was imagining `response` as a catch-all... you could put it outside this success/failure check and rename it "message" if that sounds better

**Comment by darkdarkdragon** [comment]
@wltsmrz I've made changes, please take a look - if this is what you wanted, I'll sqash commits

**Comment by wltsmrz** [comment]
LGTM

**Comment by clark800** [comment]
It looks like it's possible that a transaction could be submitted twice. If the first one succeeds, then the second one will fail because it has the same sequence number, but the user will only see the failure result and may mistakenly conclude that they have to resubmit it.

One option could be to check for `tefALREADY` errors in the reconnect case and convert them to success responses.

**Comment by darkdarkdragon** [comment]
ok, I'll make the change

**Comment by wltsmrz** [comment]
There is already reconnect logic in transactionmanager

**Comment by wltsmrz** [comment]
Probably wouldn't be a tefALREADY either, isn't that for same ledger requests?

**Comment by darkdarkdragon** [comment]
Maybe I'm wrong, but I don't transactionmanager being used in new `RippleApi`

**Comment by wltsmrz** [comment]
Maybe not

**Comment by clark800** [comment]
The docs say `tefALREADY` means "The same exact transaction has already been applied."

**Pull request merged**